### PR TITLE
ESM/CJS Exports addition 

### PR DESCRIPTION
**Pull request by barelyhuman**
related: https://github.com/pmndrs/zustand/pull/507 and https://github.com/pmndrs/zustand/pull/537

### CODE REVIEW

**Comment by barelyhuman** [comment]
**Working Sandboxes with the demo code**


Sandbox|Link
---|---
React| [CSB](https://codesandbox.io/s/react-forked-sy0o0?file=/src/App.js:0-551)
React TS| [CSB](https://codesandbox.io/s/react-typescript-forked-i8rq6)
Browserify| [CSB](https://codesandbox.io/s/react-browserify-forked-ishl6?file=/src/index.js)
Snowpack| [CSB](https://codesandbox.io/s/react-snowpack-forked-dzdk6?file=/src/index.jsx)

**Comment by dai-shi** [comment]
With this non important change https://github.com/pmndrs/zustand/pull/507#discussion_r688914744, I think it's ok for us to merge this.

**Review by dai-shi — COMMENTED** [review]
looks good.

**Comment by dai-shi** [comment]
@barelyhuman Do you think we can go with this?

**Comment by barelyhuman** [comment]
> Do you think we can go with this?

You can review and go ahead

---
Sandbox | Link
---|---
Browserify | [External Link](https://codesandbox.io/s/react-browserify-forked-9dzhs)
Snowpack | [External Link](https://codesandbox.io/s/react-snowpack-forked-gqz2m?file=/src/index.jsx)
Nextjs | [External Link](https://codesandbox.io/s/next-js-forked-edig8)

File: package.json [hunk 1]
```diff
@@ -111,7 +107,7 @@
     "test": "jest && jest --setupFiles ./tests/setProviderLessMode.ts",
     "test:dev": "jest --watch --no-coverage",
     "test:coverage:watch": "jest --watch",
-    "copy": "shx cp -r dist/src/* dist/esm && shx mv dist/src/* dist && shx rm -rf dist/{src,tests} && downlevel-dts dist dist/ts3.4 && shx cp package.json readme.md LICENSE dist && json -I -f dist/package.json -e \"this.private=false; this.devDependencies=undefined; this.optionalDependencies=undefined; this.scripts=undefined; this.prettier=undefined; this.jest=undefined; this['lint-staged']=undefined;\""
+    "copy": "shx mv dist/src/* dist && shx rm -rf dist/{src,tests} && downlevel-dts dist dist/ts3.4 && shx cp package.json readme.md LICENSE dist && json -I -f dist/package.json -e \"this.private=false; this.devDependencies=undefined; this.optionalDependencies=undefined; this.scripts=undefined; this.prettier=undefined; this.jest=undefined; this['lint-staged']=undefined;\";"
```
Comment (dai-shi) [inline package.json:110]:
```suggestion
    "copy": "shx mv dist/src/* dist && shx rm -rf dist/{src,tests} && downlevel-dts dist dist/ts3.4 && shx cp package.json readme.md LICENSE dist && json -I -f dist/package.json -e \"this.private=false; this.devDependencies=undefined; this.optionalDependencies=undefined; this.scripts=undefined; this.prettier=undefined; this.jest=undefined; this['lint-staged']=undefined;\""
```

**Review by dai-shi — APPROVED** [review]

**Pull request merged**